src/components/Brewery/Brewery.js

```diff
@@ -28,18 +52,50 @@ const MapWrapper = styled.section`
 
 const Brewery = (props) => {
 
+  const handleFavorites = () => {
+    if (props.favorites.length > 0) {
+      props.favorites.forEach(favorite => {
+        if (favorite.id == props.pub.id) {
+          props.setFavorites(props.pub);
+        } else {
+          props.setFavorites(props.pub);
+        }
+      })
+    } else {
+      props.setFavorites(props.pub);
+    }
+  }
+
+  const checkFavorites = () => {
+    if (props.favorites.length > 0) {
+      props.favorites.forEach(favorite => {
+        if (favorite.id == props.pub.id) {
+          return false;
+        } else {
+          return true;
+        }
+      })
+    } else {
+      return true;
+    }
+  }
+
   return (
     <Wrapper>
-      <h3>{props.name}</h3>
+      <h3>{props.pub.name}</h3>
       <section>
-        <p>Brewery Type: {props.brewery_type}</p>
-        <p>Address: {props.street}</p>
-        <p>{props.city}, {props.state}</p>
-        <p>Phone Number: {props.phone}</p>
+        <p>Brewery Type: {props.pub.brewery_type}</p>
+        <p>Address: {props.pub.street}</p>
+        <p>{props.pub.city}, {props.pub.state}</p>
+        <p>Phone Number: {props.pub.phone}</p>
       </section>
-      <MapWrapper>
-
-      </MapWrapper>
+      <ButtonWrapper>
+        <button
+          onClick={handleFavorites}>
+          {checkFavorites() ? 'Favorite this Brewery' : 'Remove from Favorites'}
```

Comment:
On-click immediately changes the button and adds / removes brewery from state, and the actual rendering of the button (checkFavorites) runs when the component is mounted to decide what to render within the button.